### PR TITLE
api: fix health status with tls enabled

### LINKED ISSUE #968: api[ pd/health ] not work with https
```
curl --cacert pdca/ca.pem --cert pdca/pd-server.pem --key pdca/pd-server-key.pem   https://172.16.20.2:2379/pd/health
[
  {
    "name": "pd3",
    "member_id": 7843861206316908817,
    "client_urls": [
      "https://172.16.20.6:2379"
    ],
    "health": false
  },
  {
    "name": "pd2",
    "member_id": 8566316260603844050,
    "client_urls": [
      "https://172.16.20.4:2379"
    ],
    "health": false
  },
  {
    "name": "pd1",
    "member_id": 9089730616910707339,
    "client_urls": [
      "https://172.16.20.2:2379"
    ],
    "health": false
  }
]
```

### PR DESCRIPTION
**Pull request by Connor1996**
this pd does:
 - init http client for handler in main function. Fix #968
 - reuse dialClient in redirector.go
 - disable keep alive

### CODE REVIEW

File: server/api/store.go [hunk 1]
```diff
@@ -28,12 +28,14 @@ import (
 	"github.com/unrolled/render"
 )
 
-type metaStore struct {
+// MetaStore contains meta information about a store.
+type MetaStore struct {
```
Comment (siddontang) [inline server/api/store.go]: why public heere?

File: server/api/server.go [hunk 2]
```diff
@@ -38,5 +38,10 @@ func NewHandler(svr *server.Server) http.Handler {
 
 	engine.UseHandler(router)
 
+	tlsConfig, _ := svr.GetSecurityConfig().ToTLSConfig()
```
Comment (tennix) [inline server/api/server.go]: ignore error?

File: server/api/store.go (hunk 1 above)
Comment (Connor1996) [reply server/api/store.go]: tidb_exporter need these structs to unmarshal

File: server/api/health.go [hunk 3]
```diff
@@ -56,7 +56,7 @@ func (h *healthHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			Health:     true,
 		}
 		for _, cURL := range member.ClientUrls {
-			if err := doGet(fmt.Sprintf("%s%s%s", cURL, apiPrefix, pingAPI)); err != nil {
+			if _, err := doGet(fmt.Sprintf("%s%s%s", cURL, apiPrefix, pingAPI)); err != nil {
```
Comment (disksing) [inline server/api/health.go]: I think the `resp` need Close.

**Comment by disksing** [comment]
LGTM.

**Pull request merged**